controllers/news.controller.js

```diff
@@ -51,3 +51,19 @@ const checkPropertiesOfEndpoints = (endpointsObject) => {
 
   return !pathsNeedToUpdate.length ? false : pathsNeedToUpdate;
 };
+
+exports.getArticleById = (req, res, next) => {
+  const { article_id } = req.params;
+  selectArticleById(article_id)
+    .then(({ rows }) => {
+      if (!rows.length) {
+        res.status(404).send({ msg: "not found" });
+      }
+      res.status(200).send({ article: rows[0] });
+    })
+    .catch((err) => {
+      if (err.code === "22P02") {
```

Comment:
As you go through you might want to refactor this into a custom error handler, see the notes.